yt/frontends/exodus_ii/data_structures.py

```diff
@@ -336,7 +336,30 @@ def _read_connectivity(self):
         connectivity = []
         with self._handle.open_ds() as ds:
             for i in range(self.parameters["num_meshes"]):
-                connectivity.append(ds.variables["connect%d" % (i + 1)][:].astype("i8"))
+                var = ds.variables["connect%d" % (i + 1)][:].astype("i8")
+                try:
+                    elem_type = var.elem_type.lower()
+                    if elem_type == "nsided":
+                        arbitrary_polyhedron = True
+                    elif elem_type == "nfaced":
+                        raise NotImplementedError(
+                            "3D arbitrary polyhedra are not implemented yet"
+                        )
+                    else:
+                        arbitrary_polyhedron = False
+                except AttributeError:
+                    arbitrary_polyhedron = False
+
+                conn = var[:].astype("i8")
+                if arbitrary_polyhedron:
+                    nodes_per_element = ds.variables["ebepecnt%d" % (i + 1)]
```

Comment:
We normally use fstrings for string formatting now (though this is more recent than the opening of this PR !)
```suggestion
                    nodes_per_element = ds.variables[f"ebepecnt{i + 1}"]
```

Reply:
Yes, fstrings are nice. I normally avoid them for compatibility with Python < 3.6 (and even Python 2 in the past), but if that's the yt style, I'm perfectly OK with that.